### PR TITLE
Implemented truncated list and modified guidance

### LINKED ISSUE #2111: NameValueList - Add example for displaying a list of items in vertical list as a value.
This is follow up to an office hours question regarding how to treat “services list as a value” in a name value pair. 

This can either be added (not replace) to this example, or be presented as "List Truncation" or something similar.

Here’s a [mockup](https://codesandbox.io/s/xenodochial-carlos-9qrb7?file=/src/App.js).

Re: Truncation Guidance

I think we would want to provide guidance for the number of items in this case.
Something along the lines of “Use truncation if the number of items is greater than X.” 

- If I have 6 items or less, display all items and do not use truncation.
- If I have 7+ items, use truncation, displaying 3 items, then show all button.

### PR DESCRIPTION
**Pull request by halocline**
[Deploy Preview](https://deploy-preview-2125--keen-mayer-a86c8b.netlify.app/components/namevaluelist#content-truncation)

#### What does this PR do?

- Closes #2111 
- Implemented truncated list and modified guidance

#### Where should the reviewer start?

NameValueListMultipleDefaultExample.js

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [x] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [x] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [x] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### What are the relevant issues?

#2111

#### Should this PR be mentioned in Design System updates?

Yes.

#### Is this change backwards compatible or is it a breaking change?

Backwards Compatible.

### CODE REVIEW

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 1]
```diff
@@ -155,7 +155,13 @@ In some cases, NameValuePairs may be used as read-only display for text inputs t
 
 ### Content Truncation
 
-When a value content exceeds 192 characters, truncate content and include a Button to expand to see more.
+When presenting a list of items, the list may be truncated and 
+accompanied by a Button to expand the list, displaying all items.
```
Comment (bahriaditi) [inline aries-site/src/pages/components/namevaluelist.mdx:159]: When presenting a value as a list of items, the list may be truncated and accompanied by a Button to expand it, displaying all items.

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 2]
```diff
@@ -155,7 +155,13 @@ In some cases, NameValuePairs may be used as read-only display for text inputs t
...
+
+When delimitted, content exceeding 192 characters should be truncated.
```
Comment (bahriaditi) [inline aries-site/src/pages/components/namevaluelist.mdx:161]: <del>delimitted</del>
delimited

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 3]
```diff
@@ -155,7 +155,13 @@ In some cases, NameValuePairs may be used as read-only display for text inputs t
...
 ### Content Truncation
 
-When a value content exceeds 192 characters, truncate content and include a Button to expand to see more.
+When presenting a list of items, the list may be truncated and 
+accompanied by a Button to expand the list, displaying all items.
+
+When delimitted, content exceeding 192 characters should be truncated. 
+
+As a vertical list, a list of 7+ items should be truncated, displaying the first 3 items
+with a "Show all" button; 6 or fewer items, show all items and do not truncate.
```
Comment (bahriaditi) [inline aries-site/src/pages/components/namevaluelist.mdx:165]: Trying to simplify the language a little (feel free to take it or leave it as you see fit):

For vertical lists:
- 7+ items should be truncated, displaying the first 3 items with a "Show all" button.
- For 6 or fewer items, show all items and do not truncate.

**Comment by bahriaditi** [comment]
This looks good overall – thanks @halocline!
I added a few comments on the documentation. Example looks good 👍🏼

File: aries-site/src/examples/components/namevaluelist/Examples/NameValueListReadMoreExample.js [hunk 4]
```diff
@@ -1,35 +1,104 @@
...
-          </NameValuePair>
-        ))}
+          </>
+        </NameValuePair>
       </NameValueList>
     </Box>
   );
 };
+
+const TruncateButton = ({ showAll, setShowAll }) => (
```
Comment (britt6612) [inline aries-site/src/examples/components/namevaluelist/Examples/NameValueListReadMoreExample.js:95]: nice clean up!

File: aries-site/src/pages/components/namevaluelist.mdx (hunk 3 above)
Comment (halocline) [reply aries-site/src/pages/components/namevaluelist.mdx:166]: Done.

File: aries-site/src/pages/components/namevaluelist.mdx (hunk 2 above)
Comment (halocline) [reply aries-site/src/pages/components/namevaluelist.mdx]: Done.

File: aries-site/src/pages/components/namevaluelist.mdx (hunk 1 above)
Comment (halocline) [reply aries-site/src/pages/components/namevaluelist.mdx]: Good call. Done.

File: aries-site/src/pages/components/namevaluelist.mdx [hunk 5]
```diff
@@ -155,7 +155,14 @@ In some cases, NameValuePairs may be used as read-only display for text inputs t
 
 ### Content Truncation
 
-When a value content exceeds 192 characters, truncate content and include a Button to expand to see more.
+When presenting a value as a list of items, the list may be truncated and 
+accompanied by a Button to expand it, displaying all items.
+
+When delimited, content exceeding 192 characters should be truncated. 
+
+For vertical lists:
```
Comment (taysea) [inline aries-site/src/pages/components/namevaluelist.mdx:163]: Should we provide any guidance about when to split with a "," vs stacking vertically? Right now it looks like when it's plain text it should be a comma delimited list, but when it's an Anchor is can be vertically stacked but I'm not sure that's the guidance. Is it up to the team to use their discretion to decide? @bahriaditi

Comment (bahriaditi) [reply aries-site/src/pages/components/namevaluelist.mdx:163]: Great point @taysea! From a UX perspective, I don't have enough context to provide that design guidance just yet. It'll be interesting to see new use cases that might come up, so we can start to see a pattern emerging. For the time-being, though, I'd say its up to the designer's discretion.

Comment (taysea) [reply aries-site/src/pages/components/namevaluelist.mdx:163]: pulling from Slack discussion: "short answer is - i think we should leave it to the team's discretion until we see more use cases emerging"

**Review by taysea — APPROVED** [review]
LGTM! Thanks for updating.

**Review by britt6612 — APPROVED** [review]

**Pull request merged**